client/components/deposits-overview/recent-deposits.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { Flex, FlexItem, Icon } from '@wordpress/components';
+import { calendar } from '@wordpress/icons';
+
+/**
+ * Internal dependencies.
+ */
+import strings from './strings';
+import './style.scss';
+import DepositStatusChip from 'components/deposit-status-chip';
+import { getDepositDate } from 'deposits/utils';
+import { CachedDeposit } from 'wcpay/types/deposits';
+import { useDeposits } from 'wcpay/data';
+import { formatCurrency } from 'wcpay/utils/currency';
+import { Link } from '@woocommerce/components';
+import { getDetailsURL } from 'wcpay/components/details-link';
+
+interface DepositRowProps {
+	deposit: CachedDeposit;
+}
+interface RecentDeposits {
+	deposits: CachedDeposit[];
+	isLoading: boolean;
+}
+interface RecentDepositsProps {
+	currency: string | undefined;
+}
+
+const tableClass = 'wcpay-deposits-overview__table';
+
+const GetRecentDeposits = ( currency?: string ): RecentDeposits => {
```

Comment:
We should move this to the `deposits-overview/utils.ts` to keep this file with just the components

Also small nit-pick, I think this function should start with lowercase

Reply:
🙌  This looks to me like a custom React hook, so it should use the [`use` naming convention](https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use).

e.g. `useRecentDeposits`

Reply:
🙌  This looks to me like a custom React hook (its calling another hook, `useDeposits`), so it should use the [`use` naming convention](https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use).

```suggestion
const useRecentDeposits = ( currency?: string ): RecentDeposits => {
```

Reply:
I thought so too, a linter failed me on it with lowercase, maybe due to the naming?

```
ESLint: React Hook "useDeposits" is called in function "getRecentDeposits" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter.(react-hooks/rules-of-hooks)
```

Reply:
Nice catch! Interesting that that config missed this when named `Get` rather than `get`. I wonder if this is intentional and it assumed `Get` is a react component function.

<img width="1726" alt="image" src="https://user-images.githubusercontent.com/3147296/229954440-25aa4fcc-3b49-43b0-a9f2-12278d5e13cd.png">

Reply:
> I thought so too, a linter failed me on it with lowercase, maybe due to the naming?

> ESLint: React Hook "useDeposits" is called in function "getRecentDeposits" that is neither a React function component nor a custom React Hook function. 

I think the linter rule AND @Jinksi are saying the same thing - we need to be clear about whether this is a component, or a custom hook. This looks like a custom hook to me too, so we should follow the linter recommendation and make it a true custom hook.